test/unit/uk/gov/hmrc/exports/services/mapping/declaration/TypeCodeBuilderSpec.scala

```diff
@@ -16,29 +16,73 @@
 
 package uk.gov.hmrc.exports.services.mapping.declaration
 
+import org.scalatest.{Assertion, GivenWhenThen}
 import testdata.ExportsDeclarationBuilder
 import uk.gov.hmrc.exports.base.UnitSpec
-import uk.gov.hmrc.exports.models.declaration.AdditionalDeclarationType
+import uk.gov.hmrc.exports.models.declaration.AdditionalDeclarationType.STANDARD_PRE_LODGED
+import uk.gov.hmrc.exports.models.declaration.{Country, ExportsDeclaration}
 import wco.datamodel.wco.dec_dms._2.Declaration
 
-class TypeCodeBuilderSpec extends UnitSpec with ExportsDeclarationBuilder {
+class TypeCodeBuilderSpec extends UnitSpec with ExportsDeclarationBuilder with GivenWhenThen {
+
+  val typeCodeBuilder = new TypeCodeBuilder()
+  val declarationType = STANDARD_PRE_LODGED
+
+  val modifiers = Array(withAdditionalDeclarationType(declarationType))
 
   "TypeCodeBuilder" should {
-    val builder = new TypeCodeBuilder()
 
-    "Build then add from ExportsDeclaration" in {
-      val declaration = new Declaration
-      val model = aDeclaration(withDispatchLocation("EX"), withAdditionalDeclarationType(AdditionalDeclarationType.SUPPLEMENTARY_SIMPLIFIED))
-      builder.buildThenAdd(model, declaration)
+    s"set typeCode in Declaration to 'EX${declarationType}'" when {
+
+      "dispatchLocation in ExportsDeclaration is undefined" in {
+        And("the destination country is not 'GG' or 'JE' or 'Jersey' or 'Guernsey'")
+        testTypeCode("EX", modifiers)
+      }
 
-      declaration.getTypeCode.getValue must be("EXY")
+      "dispatchLocation in ExportsDeclaration is 'EX'" in {
+        And("the destination country is not 'GG' or 'JE' or 'Jersey' or 'Guernsey'")
+        testTypeCode("EX", withDispatchLocation("EX") +: modifiers)
+      }
+    }
+
+    s"set typeCode in Declaration to 'CO${declarationType}'" when {
+
+      "dispatchLocation in ExportsDeclaration is undefined and the destination country is 'GG'" in {
+        testTypeCode("CO", withDestinationCountry(Country(Some("GG"))) +: modifiers)
+      }
+
+      "dispatchLocation in ExportsDeclaration is 'EX' and the destination country is 'GG'" in {
+        testTypeCode("CO", withDispatchLocation("EX") +: withDestinationCountry(Country(Some("GG"))) +: modifiers)
+      }
```

Comment:
Nit: To be entirely sure, you may have the same test case for other countries as well. I am not suggesting copy-pasting it over and over again. It would be more appropriate to for e.g. have a list of all countries and another list with `dispatchLocation`, and then use for-comprehension to run all combinations in a loop.

Reply:
I was just following the pattern I have seen most used in the other tests here, where a unit test covers a specific case. No problem to group them. But I'll wait your confirmation about the possible values of `locations.destinationCountry.code`